action.php

```diff
@@ -68,6 +68,28 @@ function _handle_function_submit(&$event, $param) {
             msg($this->getlang('email_err'), -1);
             $err=true;
         }
+
+        $response = $_POST["g-recaptcha-response"];
+        $url = 'https://www.google.com/recaptcha/api/siteverify';
+        $data = array(
+            'secret' => 'YOUR_SECRET',
+            'response' => $_POST["g-recaptcha-response"]
+        );
+        $options = array(
+            'http' => array (
+                'method' => 'POST',
+                'content' => http_build_query($data)
+            )
+        );
+        
+        $context  = stream_context_create($options);
+        $verify = file_get_contents($url, false, $context);
+        $captcha_success=json_decode($verify);
+        if ($captcha_success->success==false) {
+            $err=true;
+            msg ($this->getlang('recaptcha_err'),-1);
+        } else if ($captcha_success->success==true) {
```

Comment:
You forgot to invite the slack user here in this part

Reply:
The successful captcha allows the code to pass to the next stage and does not send an error, this is how the rest of the code is structured.